mycore-base/src/main/java/org/mycore/datamodel/niofs/MCRVersionedPath.java

```diff
@@ -108,14 +108,12 @@ public String toRelativePath() {
 
     @Override
     public boolean equals(Object obj) {
-        boolean result;
-        if (!(obj instanceof MCRVersionedPath that) || !getFileSystem().equals(that.getFileSystem())
-            || !this.path.equals(that.path) || !this.getOwner().equals(that.getOwner())) {
-            result = false;
-        } else {
-            result = this.hasSameVersion(that);
-        }
-        return result;
+        return
```

Comment:
Das ist sicher deutlich besser als zuvor (und wär auch ok für mich), ich würde trotzdem die alte Variante präferieren.
1. Unübersichtlich: ich sehe einen riesigen Block an Code und muss genau hinstarren. Es ist ein bisschen wie bei der streaming API. Sieht erstmal cool aus, aber ist nicht so leserlich.
2. Debugging: Ich glaube Intellij kann auch einzelne Ausdrücke debuggen, ich weiß aber nicht ob das jede IDE so unterstützt. Falls nicht, stehe ich vor einem riesigen return statement und weiß nicht warum er false oder true zurück gibt.
3. Boolsche Operationen im Kopf: Ich bin faul, sobald ich && oder || sehe hab ich schon weniger Lust mir den Code anzuschauen. In dem Fall mag das noch einfach sein. Aber selbst hier muss ich genau hinstarren um zu sehen ob auch wirklich alles mit && verknüpft ist, ob es keine klammern gibt, ob sich nicht doch irgendwo ein ! versteckt hat.